react/components/BuyButton/index.js

```diff
@@ -116,20 +128,36 @@ export const BuyButton = ({
         }
         const mutationRes = await orderFormContext.addItem({ variables })
         const { items } = mutationRes.data.addItem
+
         success = skuItems.filter(
           skuItem => !!items.find(({ id }) => id === skuItem.skuId)
         )
         await orderFormContext.refetch().catch(() => null)
       }
 
-      success =
-        success ||
-        (linkStateItems &&
-          skuItems.filter(
-            skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
-          ))
+      const addedItem =
+        linkStateItems &&
+        skuItems.filter(
+          skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
+        )
+
+      const foundItem =
+        orderFormItems &&
+        orderFormItems.filter(item => item.id === addedItem[0].skuId).length > 0
```

Comment:
why only item[0]?

Reply:
Wow! That should probably not be a `[0]`... but I don't know in which case the `addedItem` would be an array with length greater than 1. Would that be the case with Assembly Options?

Reply:
```
const addedItem =
        linkStateItems &&
        skuItems.filter(
          skuItem => !!linkStateItems.find(({ id }) => id === skuItem.skuId)
        )
```

This is how `addedItem` is defined. For it to contain more than one item, it would require `linkStateItems` to contain two items with the same skuId, and I'm aaalmost sure that should not happen.

Reply:
no problem, [0] is ok